datajoint/user_tables.py

```diff
@@ -152,6 +157,75 @@ class Lookup(UserTable):
     )
 
 
+class Params(UserTable):
+    """
+    Inherit from this class if the table's values are paramsets. Adds hashing of fields
+    that include 'param' and stores as additional unique index 'paramset hash'
+    """
+
+    _prefix = "##"
+    tier_regexp = (
+        r"(?P<params>" + _prefix + _base_regexp.replace("TIER", "params") + ")"
+    )
+
+    def __init__(self):
+        super().__init__()
+        if "params" not in self.definition:
+            raise DataJointError("Params tables must have at least one 'params' field")
+        self.definition += "__paramset_hash: uuid\nunique index (__paramset_hash)\n"
+
+    def insert(self, rows, **kwargs):
+        """
+        Insert rows into Params table, ensuring uniqueness for param fields.
+
+        Note: does not accept QueryExpressions as rows. Accepts all other parameters of
+        standard dj.table.insert function including replace, skip_duplicates, etc.
+        """
+
+        if isinstance(rows, QueryExpression):
+            raise NotImplementedError(
+                "Params tables do not yet support insertion via QueryExpression"
+            )
+
+        rows = self._normalize_insert_data(rows)
+
+        new_rows = []
+        for row in rows:
+            if isinstance(row, np.record):
+                row_values = tuple(row)
+            elif isinstance(row, collections.abc.Mapping):
+                row_values = row.values()
+            else:
+                row_values = row
+
+            paramset_dict = {
+                field: value
+                # zip_longest accounts for nullable params fields
+                for field, value in zip_longest(
```

Comment:
I advise against using `zip_longest` since this will likely lead to errors. When inserting records without naming the attributes, then all values must be provided.